### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by pwnsdx**
- Using const instead of let for "argv"
- Separated defines
- Use ` ${}` instead of ' '+
- Added comments

### CODE REVIEW

File: electron/js/config.js [hunk 1]
```diff
@@ -55,6 +55,47 @@ let config = {
   UPDATE_DELAY: 5 * 60 * 1000,
   UPDATE_INTERVAL: 24 * 60 * 60 * 1000,
 
+  EMBED_DOMAINS: [
+
+    // Youtube
+    {
+      name: 'Youtube',
```
Comment (lipis) [inline electron/js/config.js]: `YouTube`

File: electron/js/config.js [hunk 2]
```diff
@@ -55,6 +55,47 @@ let config = {
...
+      hostname: ['www.youtube-nocookie.com'],
+      allowedExternalLinks: [
+        'www.youtube.com',
+      ],
+    },
+
+    // Vimeo
```
Comment (lipis) [inline electron/js/config.js]: no need for comments here, since the next line is showing that

File: package.json [hunk 3]
```diff
@@ -38,5 +38,8 @@
     "grunt-git": "1.0.3",
     "grunt-github-changes": "0.1.0",
     "load-grunt-tasks": "3.5.2"
+  },
+  "dependencies": {
+    "debug": "^2.6.8"
```
Comment (lipis) [inline package.json]: remove `^`

File: electron/main.js [hunk 4]
```diff
@@ -385,3 +409,81 @@ fs.stat(consoleLog, function(err, stats) {
...
+        if (!util.isMatchingEmbed(_url)) {
+          e.preventDefault();
+          webviewProtectionDebug('Prevented to show an unauthorized <webview>. URL: %s', _url);
+        }
+      });
+    });
+  }
+}
+
+(new ElectronWrapperInit()).run();
```
Comment (lipis) [inline electron/main.js]: end of line

File: electron/main.js [hunk 5]
```diff
@@ -200,6 +213,9 @@ function showMainWindow() {
       'backgroundThrottling': false,
       'nodeIntegration': false,
       'preload': PRELOAD_JS,
+
+      // Enable <webview>
```
Comment (lipis) [inline electron/main.js]: no need for comment.. `webviewTag: true` is self explanatory

File: electron/main.js [hunk 6]
```diff
@@ -19,32 +19,50 @@
...
+/*eslint-disable no-unused-vars*/
+const USER_DATAS_PATH = app.getPath('userData');
+/*eslint-enable no-unused-vars*/
+
+// Local files defines
+const PRELOAD_JS = path.join(APP_PATH, 'js', 'preload.js');
+const WRAPPER_CSS = path.join(APP_PATH, 'css', 'wrapper.css');
+const SPLASH_HTML = `file://${path.join(APP_PATH, 'html', 'splash.html')}`;
+const CERT_ERR_HTML = `file://${path.join(APP_PATH, 'html', 'certificate-error.html')}`;
+const ABOUT_HTML = `file://${path.join(APP_PATH, 'html', 'about.html')}`;
```
Comment (ffflorian) [inline electron/main.js]: Why reordering the variables?

File: electron/js/config.js [hunk 7]
```diff
@@ -55,6 +55,39 @@ let config = {
...
+    {
+      name: 'Vimeo',
+      hostname: ['player.vimeo.com'],
+      allowedExternalLinks: [
+        'vimeo.com',
+        'player.vimeo.com',
+      ],
+    },
+    {
+      name: 'Soundcloud',
```
Comment (ffflorian) [inline electron/js/config.js]: SoundCloud

File: electron/main.js [hunk 8]
```diff
@@ -270,9 +290,17 @@ function showMainWindow() {
...
 
-  main.webContents.on('new-window', function(event, url) {
+  // Handle the new window event in the main Browser Window
+  main.webContents.on('new-window', (event, _url) => {
     event.preventDefault();
-    shell.openExternal(url);
+
+    // Ensure the link does not come from a webview
+    if (typeof event.sender.viewInstanceId !== 'undefined') {
+      debugMain('New window did came from a webview, aborting.');
```
Comment (ffflorian) [inline electron/main.js]: `New window was created from a webview, aborting.`

**Pull request merged**